### PR TITLE
maintenance: updates  for 2.2.0 release

### PR DESCRIPTION
**Pull request by tpike3**
update `__init__.py`
update `HISTORY.md`
fix ruff issue `agent.py`

### CODE REVIEW

File: HISTORY.md [hunk 1]
```diff
@@ -2,6 +2,44 @@
 title: Release History
 ---
 
+# 2.2.0 (2024-01-08)
+
+# Highlights
+This release incorporates several majors changes to Mesa. Notably, this provides two experimental features. (1) It reimagines Mesa's agent handling to have agents in the model class and store them in sets. The intent is to make Mesa more efficient and give users more control over agent scheduling to allow for innumerable options. (2) It adds property layer and property grid so users can layer on multiple grids making Mesa and Geo-Mesa more similar.
```
Comment (rht) [inline HISTORY.md:8]: Needs to link to the feedbacks: https://github.com/projectmesa/mesa/discussions/1919 and https://github.com/projectmesa/mesa/discussions/1932.

**Comment by EwoutH** [comment]
I also wrote a highlights section in [#1917 ](https://github.com/projectmesa/mesa/pull/1917#issuecomment-1873504101), if you want you can incorporate some of [it](https://gist.github.com/EwoutH/4cb91f6661239280eed19d6c36e81b46).

**Comment by jackiekazil** [comment]
After this one change, I think we are good to go -- all please speak up ... if it is minor-ish, I am going to say make it PR to get this out the door.

**Review by EwoutH — APPROVED** [review]

File: HISTORY.md (hunk 1 above)
Comment (rht) [reply HISTORY.md]: I amended the commit to include this.

**Review by rht — APPROVED** [review]

**Comment by tpike3** [comment]
I updated the date to the 9th... as it was approved twice I am merging. 

@EwoutH  that is a phenomenal write up! Just to get this out and as the changes are already approved, I will use that for the email to the community

**Comment by jackiekazil** [comment]
@EwoutH are u trying to get auto release working? I noticed the "ignore" tag. Let me know or I am planning on release tonight manually.

**Comment by EwoutH** [comment]
Tom already released 2.2.0! I'm just doing some deferred maintenance.

I reordered and added the extended highlights section in the [2.2.0 release notes](https://github.com/projectmesa/mesa/releases/tag/v2.2.0) by the way.

**Pull request merged**